### PR TITLE
fix(dashboards): Widget Viewer path name match to only check digits

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Updates `checkIfShouldMountWidgetViewerModal` to more accurately match widget ids in the pathname:
`/\/widget\/[0-9]+\/$/`

This should only match pathnames that end with something like `/widget/123/`.
This fixes the case where it was matching `/widget/new/` since `new` is not a number
This should not be matching `/widget/0/edit/` due to the `$` token

### CODE REVIEW

**Review by shruthilayaj — APPROVED** [review]

**Comment by edwardgou-sentry** [comment]
Hey @priscilawebdev , I think this might be enough to fix the pathname clashes. We're only checking for numbers now and the pathname must end with `/widget/${widgetId}/` Let me know what you think!

**Review by priscilawebdev — APPROVED** [review]
can confirm that so long the URL doesn't have a query string it works.

thanks for fixing it 👏

File: static/app/views/dashboardsV2/detail.tsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ class DashboardDetail extends Component<Props, State> {
       location,
       router,
     } = this.props;
-    if (location.pathname.match(/\/widget\/\w*\/$/) && !this.isWidgetBuilderRouter) {
+    if (location.pathname.match(/\/widget\/[0-9]+\/$/) && !this.isWidgetBuilderRouter) {
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/detail.tsx:117]: do we want to remove the second check?

Comment (edwardgou-sentry) [reply static/app/views/dashboardsV2/detail.tsx]: 👍 removed!

**Pull request merged**